crates/rb-sys-test-helpers/src/ruby_test_executor.rs

```diff
@@ -141,6 +142,9 @@ pub unsafe fn setup_ruby_unguarded() {
         rb_sys::rb_w32_sysinit(&mut argc, &mut argv);
     }
 
+    let mut stack_marker: VALUE = 0;
+    ruby_init_stack(addr_of_mut!(stack_marker) as *mut _);
```

Comment:
This was causing GC issues in Ruby 3.4, due to Ruby being unable to mark objects on the machine stack. Not sure why this wasn't an issue before, but now it's fixed.